crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -1917,7 +1917,7 @@ impl Backend {
         ) = NamedChain::try_from(self.env.read().env.cfg.chain_id)
         {
             // Block number is the best number.
-            block.header.number = self.best_number();
+            block.header.number = number;
```

Comment:
I think the proper fix would be
```rust
            if self.is_fork() {
                // Block number is the best number.
                block.header.number = self.best_number();
            }
```

Reply:
Hmm. Wouldn't that cause the same issue in fork mode? 

E.g:
If anvil was forked at `X` block number and then two blocks have been mined i.e `self.best_number() == X + 2`. 

We call `eth_getBlockByNumber(X)`, the resultant `block.number` field would be `X + 2` whereas it should be `X`.

Reply:
Am I missing something?

Reply:
hm, I guess you're right, should be the same